proto/redpanda/api/console/v1alpha1/security.proto

```diff
@@ -0,0 +1,751 @@
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "The role name. i.e. the principal is `RedpandaRole:<role>`",
+      example: "\"admin\""
+    }
+  ];
+}
+
+// CreateRoleResponse is the response for CreateRole.
+message CreateRoleResponse {
+  // The role name.
+  string role = 1;
+}
+
+// CreateRoleRequest is the request for CreateRole.
+message GetRoleRequest {
+  // The role name.
+  string role = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "The role name. i.e. the principal is `RedpandaRole:<role>`",
+      example: "\"admin\""
+    }
+  ];
+}
+
+// GetRoleResponse is the response to GetRole.
+message GetRoleResponse {
+  // The name of the role.
+  string name = 1;
+
+  // The principal type.
+  Security.PrincipalType principal_type = 2;
+}
+
+// UpdateRoleRequest is the request for UpdateRole.
+message UpdateRoleRequest {
+  // The role name.
+  string role = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "The role name. i.e. the principal is `RedpandaRole:<role>`",
+      example: "\"admin\""
+    }
+  ];
+
+  // The new role name.
+  string new_name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+}
+
+// UpdateRoleResponse is the response for UpdateRole.
+message UpdateRoleResponse {
+  // The role name.
+  string role = 1;
+}
+
+// DeleteRoleRequest is the request for DeleteRole.
+message DeleteRoleRequest {
+  // The role name.
+  string role = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+
+  // Whether to delete the ACLs bound to the role.
+  optional bool delete_acls = 2;
+}
+
+// DeleteRoleResponse is the response for DeleteRole.
+message DeleteRoleResponse {}
+
+// List role members for a role. That is user principals assigned to that role.
+message ListRoleMembersRequest {
+  // Filter options.
+  message Filter {
+    // Filter results to only mamber named with the prefix.
+    string name_prefix = 1 [(buf.validate.field).string.max_len = 128];
```

Comment:
Same thought here regarding `name_prefix` vs `name_contains`. Maybe we want to have both.